bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/svdrp/SVDRPEpgEvent.java

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.vdr.internal.svdrp;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.NoSuchElementException;
+import java.util.StringTokenizer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link SVDRPEpgEvent} contains SVDRP Response Data for an EPG Event
+ *
+ * @author Matthias Klocke - Initial contribution
+ */
+@NonNullByDefault
+public class SVDRPEpgEvent {
+
+    public enum TYPE {
+        NOW,
+        NEXT
+    }
+
+    private String title = "";
+    private String subtitle = "";
+    private Date begin = new Date();
+    private Date end = new Date();
+    private int duration;
+
+    private SVDRPEpgEvent() {
+    }
+
+    /**
+     * parse object from SVDRP Client Response
+     *
+     * @param message SVDRP Client Response
+     * @return SVDRPEpgEvent Object
+     * @throws SVDRPParseResponseException thrown if response data is not parseable
+     */
+    public static SVDRPEpgEvent parse(String message) throws SVDRPParseResponseException {
+        SVDRPEpgEvent entry = new SVDRPEpgEvent();
+        StringTokenizer st = new StringTokenizer(message, System.lineSeparator());
+
+        while (st.hasMoreTokens()) {
+            String line = st.nextToken();
+            if (line.length() >= 1 && !line.startsWith("End")) {
+                switch (line.charAt(0)) {
+                    case 'T':
+                        entry.setTitle(line.substring(1).trim());
+                        break;
+                    case 'S':
+                        entry.setSubtitle(line.substring(1).trim());
+                        break;
+                    case 'E':
+                        StringTokenizer lt = new StringTokenizer(line.substring(1).trim(), " ");
+                        lt.nextToken(); // event id
+                        try {
+                            long begin = Long.parseLong(lt.nextToken());
+                            begin *= 1000L;
+                            entry.setBegin(new Date(begin));
```

Comment:
You could directly use `Instant` instead of converting it to `Date` and then to `Instant`.
```suggestion
                            long begin = Long.parseLong(lt.nextToken())
                            entry.setBegin(Instant.ofEpochSecond(begin));
```